packages/flutter_tools/lib/src/commands/build_ios.dart

```diff
@@ -130,7 +131,63 @@ class BuildIOSArchiveCommand extends _BuildIOSSubCommand {
     return super.validateCommand();
   }
 
-  Future<void> _validateXcodeBuildSettingsAfterArchive() async {
+  // Parses Contents.json into a map, with the key to be the combination of (idiom, size, scale), and value to be the icon image file name.
+  Map<String, String> _parseIconContentsJson(String contentsJsonDirName) {
+    final Directory contentsJsonDirectory = globals.fs.directory(contentsJsonDirName);
+    if (!contentsJsonDirectory.existsSync()) {
+      return <String, String>{};
+    }
+    final File contentsJsonFile = contentsJsonDirectory.childFile('Contents.json');
+    final Map<String, dynamic> content = json.decode(contentsJsonFile.readAsStringSync()) as Map<String, dynamic>;
+    final List<dynamic> images = content['images'] as List<dynamic>? ?? <dynamic>[];
+
+    final Map<String, String> iconInfo = <String, String>{};
+
+    for (final dynamic image in images) {
+      final Map<String, dynamic> imageMap = image as Map<String, dynamic>;
+      final String? idiom = imageMap['idiom'] as String?;
+      final String? size = imageMap['size'] as String?;
+      final String? scale = imageMap['scale'] as String?;
+      final String? fileName = imageMap['filename'] as String?;
+
+      if (size != null && idiom != null && scale != null && fileName != null) {
+        iconInfo['$idiom $size $scale'] = fileName;
+      }
+    }
+
+    return iconInfo;
+  }
+
+  Future<void> _validateIconsAfterArchive(StringBuffer messageBuffer) async {
+    final BuildableIOSApp app = await buildableIOSApp;
+    final String templateIconImageDirName = await app.templateAppIconDirNameForImages;
+
+    final Map<String, String> templateIconMap = _parseIconContentsJson(app.templateAppIconDirNameForContentsJson);
+    final Map<String, String> projectIconMap = _parseIconContentsJson(app.projectAppIconDirName);
+
+    // find if any of the project icons conflict with template icons
+    final bool hasConflict = projectIconMap.entries
+        .where((MapEntry<String, String> entry) {
+      final String projectIconFileName = entry.value;
+      final String? templateIconFileName = templateIconMap[entry.key];
+      if (templateIconFileName == null) {
+        return false;
+      }
+
+      final File projectIconFile = globals.fs.file(globals.fs.path.join(app.projectAppIconDirName, projectIconFileName));
+      final File templateIconFile = globals.fs.file(globals.fs.path.join(templateIconImageDirName, templateIconFileName));
+      return projectIconFile.existsSync()
+          && templateIconFile.existsSync()
+          && md5.convert(projectIconFile.readAsBytesSync()) == md5.convert(templateIconFile.readAsBytesSync());
+    })
+    .isNotEmpty;
+
+    if (hasConflict) {
+      messageBuffer.writeln('\nWarning: You may want to replace template app icons.');
```

Comment:
"template" is our internal word for what gets used when you `flutter create`, I don't think developers would know what "template app icons" means.  And "you may" doesn't match the `Action Required: You must set a build name and number in the pubspec.yaml` type wording we have elsewhere.

What about:
```
Warning: App icon is set to the default placeholder icon. Replace with unique icons.
```